modules/custom/moj_resources/src/ContentApiClass.php

```diff
@@ -106,6 +113,22 @@ private function loadNodesDetails($nid)
   private function decorateContent($node)
   {
     $content_type = $node->type->target_id;
+
+    if (($this->prison != 0) && (count($node->field_moj_prisons) > 0)) {
+      $found = false;
+
+      foreach ($node->field_moj_prisons as $key => $n) {
+        if ($this->prison == $n->target_id) {
+          $found = true;
+          break;
+        }
+      }
+
+      if (!$found) {
+        return [];
+      }
+    }
+
```

Comment:
I wonder if we could use `in_array(thing, array)` here? 

Assuming `$node->field_moj_prisons` is an array, 

We can test this with:

> `var_dump` just prints the data type out to the console in `php -a`, which is handy for debugging.

```php
$prison_list = array(
  0 => array(
    'target_id' => '123',
    'name' => 'test 1'
  ), 
  1 => array(
    'target_id' => '234',
    'name' => 'test 2'
  )
);

# extracts a nested column into a 1-dimensional array
$prison_ids = array_column($prison_list, 'target_id');

# check for presence with in_array
var_dump(in_array('123', $prison_ids));
#> bool(true)

var_dump(in_array('432', $prison_ids));
#> bool(false)

var_dump(in_array(0, $prison_ids));
#> bool(false)

var_dump(in_array(null, $prison_ids));
#> bool(false)
```

I _think_ that means we can simplify the above to:
```php
$prison_ids = array_column($node->field_moj_prisons, 'target_id')

if(!in_array($this->prison, $prison_ids)) {
  return [];
}
```

Reply:
Looks good, made the change, will push to dev and check 👍

Reply:
@spikeheap as with all things PHP this doesn't work as expected, the array functions are a bit weird.

Reply:
Actually that could be a load of beep...I'll try it again later.

Reply:
So yes, Drupal/PHP are playing weirdness here. I seem to remember I came across this on something else ages ago, the 'arrays' aren't arrays even though they look like arrays and everything says they should behave like arrays, there's something in the data structure that means PHP doesn't see them as such. So I've reverted it back to my long-winded way of doing it 😄